micromagneticdata/abstract_drive.py

```diff
@@ -0,0 +1,323 @@
+
+        Examples
+        --------
+        1. Getting initial magnetisation field.
+
+        >>> import os
+        >>> import micromagneticdata as md
+        ...
+        >>> dirname = dirname=os.path.join(os.path.dirname(__file__),
+        ...                                'tests', 'test_sample')
+        >>> drive = md.Drive(name='system_name', number=0, dirname=dirname)
+        >>> drive.m0
+        Field(...)
+
+        """
+        return df.Field.fromfile(self._m0_path)
+
+    @property
+    @abc.abstractmethod
+    def _m0_path(self):
+        """Path to m0 file."""
+
+    @property
+    @abc.abstractmethod
+    def table(self):
+        """Table object."""
+
+    @property
+    def n(self):
+        """Number of steps.
+
+        This property returns the number of rows in the drive table.
+
+        Returns
+        -------
+        int
+
+            Number of steps.
+
+        Examples
+        --------
+        1. Getting the number of steps.
+
+        >>> import os
+        >>> import micromagneticdata as md
+        ...
+        >>> dirname = dirname=os.path.join(os.path.dirname(__file__),
+        ...                                'tests', 'test_sample')
+        >>> drive = md.Drive(name='system_name', number=0, dirname=dirname)
+        >>> drive.n
+        25
+
+        """
+        return len(self.table.data.index)
+
+    @property
+    @abc.abstractmethod
+    def _step_files(self):
+        """List of filenames of individual snapshots."""
+
+    def __getitem__(self, item):
+        """Magnetisation field of an individual step.
+
+        Returns
+        -------
+        int
+
+            Step number.
+
+        Examples
+        --------
+        1. Getting the field of a particular step.
+
+        >>> import os
+        >>> import micromagneticdata as md
+        ...
+        >>> dirname = dirname=os.path.join(os.path.dirname(__file__),
+        ...                                'tests', 'test_sample')
+        >>> drive = md.Drive(name='system_name', number=0, dirname=dirname)
+        >>> drive[5]
+        Field(...)
+
+        """
+        return df.Field.fromfile(filename=self._step_files[item])
+
+    def __iter__(self):
+        """Iterator.
+
+        This iterator iterates through all magnetisation field in drive and
+        yields ``discretisedfield.Field`` objects.
+
+        Returns
+        -------
+        discretisedfield.Field
+
+            Magnetisation field.
+
+        Examples
+        --------
+        1. Iterating drive.
+
+        >>> import os
+        >>> import micromagneticdata as md
+        ...
+        >>> dirname = dirname=os.path.join(os.path.dirname(__file__),
+        ...                                'tests', 'test_sample')
+        >>> drive = md.Drive(name='system_name', number=0, dirname=dirname)
+        >>> list(drive)
+        [...]
+
+        """
+        for filename in self._step_files:
+            yield df.Field.fromfile(filename)
```

Comment:
Could you maybe attempt `yield from (df.Field.fromfile(filename) for filename in self._step_files)`?